torch/utils/_pytree.py

```diff
@@ -238,28 +257,33 @@ def _odict_unflatten(
     dict,
     _dict_flatten,
     _dict_unflatten,
+    _register_cxx_pytree_node=False,
 )
 _register_pytree_node(
     list,
     _list_flatten,
     _list_unflatten,
+    _register_cxx_pytree_node=False,
```

Comment:
Hmm, just curious, why don't we register the cxx nodes here, is it because cxx pytree already does its own registratio?

Reply:
> is it because cxx pytree already does its own registration of the default collections?

Yes. We are going to make the C++ pytree and Python pytree analog to each other. The C++ pytree currently support:

- tuple
- list
- dict
- OrderedDict
- defaultdict
- deque
- NamedTuple
- PyStructSequence

Reply:
Can we add parity tests for all of these builtin types somehow?

Reply:
> Can we add parity tests for all of these builtin types somehow?

We can do this since we do not allow registering a single type twice.

```python
for cls in builtin_types:
    with assertRaises(TypeError):
        pytree.register_pytree_node(cls, ...)
```

I will add these tests after we make the builtin types for Python pytree aligned with C++ pytree. I'll open follow-up PRs to:

- add defaultdict and deque
- add function `is_structseq` / `is_stuctseq_class` and support `torch.return_types.*` natively.